### PR TITLE
catdoc: fix build on darwin

### LINKED ISSUE #133440: catdoc: empty output on macOS
I'm not at all familiar with catdoc, but I was building a bunch of random stuff (on macOS) to test some tooling I'm working on and it coincidentally broke because the catdoc output in my store is empty.

I took a peek at hydra and it confirms that it produces output on Linux, but not on macOS:
- https://hydra.nixos.org/job/nixpkgs/trunk/catdoc.aarch64-linux/latest-finished
- https://hydra.nixos.org/job/nixpkgs/trunk/catdoc.x86_64-linux/latest-finished
- https://hydra.nixos.org/job/nixpkgs/trunk/catdoc.aarch64-darwin/latest-finished
- https://hydra.nixos.org/job/nixpkgs/trunk/catdoc.x86_64-darwin/latest-finished

The only packages I see that consume catdoc are [catdocx](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/text/catdocx/default.nix) and [recoll](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/search/recoll/default.nix), so they may both be subtly broken on macOS? catdoc doesn't have a listed maintainer, so I'm mentioning maintainers of catdocx and recoll in case they have a sense of whether it should work and just needs a fix, or should maybe get marked broken?
- recoll: @jcumming @kiyengar
- catdocx: @michalrus

### PR DESCRIPTION
**Pull request by marsam**
###### Motivation for this change

Fixes https://github.com/NixOS/nixpkgs/issues/133440 cc: @abathur 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/catdoc/default.nix [hunk 1]
```diff
@@ -16,6 +16,12 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  # Remove INSTALL file on Darwin to avoid `make` misinterpret it as up-to-date
+  # target on case-insensitive filesystems.
+  preInstall = lib.optionalString stdenv.isDarwin ''
```
Comment (Mic92) [inline pkgs/tools/text/catdoc/default.nix:21]: I don't think it has any disadvantages to do this everywhere. This way we avoid bitrot.

```suggestion
  preInstall = ''
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134141` at bcec50df run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>catdoc</li>
    <li>catdocx</li>
    <li>python38Packages.recoll</li>
    <li>recoll (python39Packages.recoll)</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/tools/text/catdoc/default.nix:31:5:
```
   |
31 |     maintainers = with maintainers; [];
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/text/catdoc/default.nix:30:5:
```
   |
30 |     license = licenses.gpl2;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/tools/text/catdoc/default.nix:13:5:
```
   |
13 |     (fetchpatch {
   |     ^
```
</li>
  </ul>
</details>

**Comment by abathur** [comment]
I can't atm, but I'll try to remember to check this out and confirm whether it worked within ~3 days :)

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 134141` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python38Packages.recoll</li>
    <li>recoll (python39Packages.recoll)</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>catdoc</li>
    <li>catdocx</li>
  </ul>
</details>

**Review by veprbl — APPROVED** [review]

**Review by risicle — APPROVED** [review]
Fixes it for me, macos 10.15, `nixpkgs-review` still happy on nixos x86_64.

`recoll` still fails on macos but that's another problem...

**Pull request merged**